tests/test_errors.py

```diff
@@ -31,3 +36,51 @@ def test_fk_not_exists(self, client, endpoint, kf_id):
         else:
             message = 'Longer than maximum length 11'
             assert message in response['_status']['message']
+
+    def test_uniqueness_constraints(self, client, entities):
+        """ Test integrity error from uniqueness violations """
+        # apparently some of our operations are destructive to ENTITY_PARAMS
+        # state used by other tests
+        global ENTITY_PARAMS
+        ENTITY_PARAMS = copy.deepcopy(ENTITY_PARAMS)
+
+        # init with dummy data
+        for endpoint, inputs in ENTITY_PARAMS['fields'].items():
```

Comment:
Is this duplicating what's in the `entities` fixture?
Would it be better to look up one of the relationships and try to add it again?

Reply:
It does try to add again below to test the constraint messages. This just makes sure that all of the necessary pieces are in place for the first one to actually be recorded, otherwise you have to deal with whatever your foreign key requirements are to get the first one in at all and I didn't think that was in scope for the testing here.  (it deepcopies ENTITY_PARAMS because otherwise _add_foreign_keys breaks subsequent tests for some reason. I'm not sure why that would be the case, but *shrug*)

Reply:
Hmm, I concur with Dan's comment. You should be able to use the entities fixture and add the same entity again. I'll try to look at this tomorrow

Reply:
Perhaps it would be better to not use the entities fixture and instead load up the minimal data manually? This would be a lot more clear in the test as well as avoid having to set up all the data in the api.

Reply:
Let's talk about this tomorrow. I need to learn more about this entities fixture of which you speak.